### PR TITLE
Upgrade to Django 1.11

### PR DESCRIPTION
**Pull request by nllong**
#### Any background context you want to provide?
Need to upgrade Django and other dependencies

#### What's this PR do?
Upgrade to Django 1.11.6
Upgrade Django Rest Framework and Swagger. There were some old API endpoints that were in the root, which have been moved to /api/v1.

**Potential breaking changes**

- Progress API (/api/v2/progress) endpoint. Before this was a POST, now it is a GET with the progress key ID being passed as the id (/api/v2/progress/<progress_key>)
- Delete File (/api/delete_file) is now part of the import_files namespace (DELETE /api/import_files/{}/)
- /api/v1/get_columns moved to GET /api/v1/columns 

There is a bit more work to cleanup swagger since it is using the new format (no YML parsing).

#### How should this be manually tested?
Run tests, monkey testing

#### What are the relevant tickets?
N/A

#### Screenshots (if appropriate)
N/A

#### Definition of Done:
- [x] Is there appropriate test coverage? (e.g. ChefSpec, Mocha/Chai, Python, etc.)
- [x] Does this PR require a Selenium test? (e.g. Browser-specific bugs or complicated UI bugs)
- [x] Does this PR require a regression test? All fixes require a regression test.
- [x] Does this add new dependencies? If so, does PIP, npm, bower requirements need to be updated?

**YES! pip install**

### CODE REVIEW

**Comment by axelstudios** [comment]
Just a note - **you have to delete \*.pyc files**

**Comment by Myoldmopar** [comment]
> Just a note - you have to delete *.pyc files

And in case you didn't know this beautiful little menuitem existed:

image

File: config/management/commands/grab_manifest.py [hunk 1]
```diff
@@ -2,45 +2,42 @@
 :copyright (c) 2014 - 2017, The Regents of the University of California, through Lawrence Berkeley National Laboratory (subject to receipt of any required approvals from the U.S. Department of Energy) and contributors. All rights reserved.  # NOQA
 :author
 """
-import optparse
```
Comment (Myoldmopar) [inline config/management/commands/grab_manifest.py]: The change from `optparse` to a standardized `add_arguments` call is :fire:.

File: config/urls.py [hunk 2]
```diff
@@ -21,12 +22,11 @@
     url(r'^app/', include('seed.urls', namespace="seed", app_name="seed")),
 
     # root configuration items
-    url(r'^eula/', include('tos.urls', namespace='tos', app_name='tos')),
     url(r'^i18n/', include('django.conf.urls.i18n')),
     url(r'^robots\.txt', robots_txt, name='robots_txt'),
 
     # API
-    url(r'^api/swagger/', include('rest_framework_swagger.urls'), name='swagger'),
+    url(r'^api/swagger/', get_swagger_view(title='SEED API'), name='swagger'),
```
Comment (Myoldmopar) [inline config/urls.py]: A much nicer approach here.

File: seed/api/v1/urls.py [hunk 3]
```diff
@@ -5,27 +5,28 @@
...
+router = routers.DefaultRouter()
+router.register(r'columns', ColumnViewSetV1, base_name="columns")
 
 urlpatterns = [
     # template routes
+    url(r'^', include(router.urls)),
+
     url(r'^search_buildings/$', search_buildings, name='search_buildings'),
-    url(r'^get_default_columns/$', get_default_columns, name='get_default_columns'),
     url(r'^set_default_columns/$', set_default_columns, name='set_default_columns'),
```
Comment (Myoldmopar) [inline seed/api/v1/urls.py]: Was there a reason this `set` method wasn't pulled into the viewset?  Should this be added to a followup todo list?

File: seed/data_importer/tests/test_case_a.py [hunk 4]
```diff
@@ -36,7 +38,12 @@ def setUp(self):
         self.fake_row = FAKE_ROW
         selfvars = self.set_up(import_file_source_type)
         self.user, self.org, self.import_file, self.import_record, self.cycle = selfvars
-        self.import_file.load_import_file(osp.join(osp.dirname(__file__), 'data', filename))
+        filepath = osp.join(osp.dirname(__file__), 'data', filename)
+        self.import_file.file = SimpleUploadedFile(
+            name=filename,
+            content=open(filepath, 'rb').read()
+        )
+        self.import_file.save()
```
Comment (Myoldmopar) [inline seed/data_importer/tests/test_case_a.py]: I assume this is due to the enhanced restrictions on resource data.  Seems good.

File: seed/landing/tests.py [hunk 5]
```diff
@@ -31,25 +29,25 @@ def test_simple_login(self):
...
-
-        # django-tos does not log the user in yet
-        self.assertFalse('_auth_user_id' in self.client.session)
-
-        # and submitting 'accept=accept' to the confirm url logs you in
-        self.client.post(self.tos_url, {'accept': 'accept'})
-
-        # now we're logged in
-        self.assertTrue('_auth_user_id' in self.client.session)
+    # def test_tos_login(self):
```
Comment (Myoldmopar) [inline seed/landing/tests.py]: Any reason to keeping this commented code around?

Comment (Myoldmopar) [inline seed/landing/tests.py]: Reason to keep this code around?

File: seed/landing/views.py [hunk 6]
```diff
@@ -52,41 +49,36 @@ def login_view(request):
                 username=form.cleaned_data['email'].lower(),
                 password=form.cleaned_data['password']
             )
-            if new_user and new_user.is_active:
+            if new_user is not None and new_user.is_active:
```
Comment (Myoldmopar) [inline seed/landing/views.py]: I'm curious why you had to add the `is not None` clause here.  Seems like this is equivalent to before, but more wordy...hmm.

File: seed/static/seed/js/services/dataset_service.js [hunk 7]
```diff
@@ -43,12 +43,11 @@ angular.module('BE.seed.service.dataset', []).factory('dataset_service', [
     };
 
     dataset_service.delete_file = function (file_id) {
-      return $http.delete(window.BE.urls.delete_file, {
+      return $http.delete('/api/v2/import_files/' + file_id + '/', {
```
Comment (Myoldmopar) [inline seed/static/seed/js/services/dataset_service.js]: I know there were really good intentions behind the window.BE.urls shortcuts, but I actually like the explicit nature of this better.  Not sure why.

File: seed/tests/test_api.py [hunk 8]
```diff
@@ -77,25 +76,25 @@ def test_get_api_endpoints_utils(self):
...
+                 '/api/v2.1/properties',)):
                 self.assertTrue(
                     url.endswith('/'),
                     "Endpoint %s does not end with / as expected" % url
                 )
 
     def test_get_api_schema(self):
         """
-        Test of 'schema' generator.
+        Test of 'schem`a' generator.
```
Comment (Myoldmopar) [inline seed/tests/test_api.py]: What?

File: config/urls.py (hunk 2 above)
Comment (nllong) [reply config/urls.py]: this was product of upgrading the swagger django rest framework.

File: seed/api/v1/urls.py (hunk 3 above)
Comment (nllong) [reply seed/api/v1/urls.py]: yeah, it should be added to a todo list. The set_ and get_ here are never used as far as I can tell.

File: seed/landing/tests.py (hunk 5 above)
Comment (nllong) [reply seed/landing/tests.py]: I kept it around to make sure it was explicit that we aren't handling TOS anymore... and haven't been for quite some time.

File: seed/landing/views.py (hunk 6 above)
Comment (nllong) [reply seed/landing/views.py]: i am just more verbose i guess. For some reason I thought there was a difference between `if new_user` and `if user is not None`

File: seed/tests/test_api.py (hunk 8 above)
Comment (nllong) [reply seed/tests/test_api.py]: oops

**Pull request merged**